torch/_dynamo/variables/tensor.py

```diff
@@ -158,7 +159,9 @@ def specialize(value: torch.Tensor):
             # already. We could remove the discrepancy here, by having ConstantVariable be more permissive for
             # constant backed SymInts, but that assert being strict has led to some good signal in hunting bugs, and
             # I'd like to keep it around for now.
-            props["size"] = tuple([int(s) for s in value.size()])
+            props["size"] = tuple(
+                [int(s) if is_symbolic(s) else s for s in value.size()]
+            )
```

Comment:
what's this change for? I imagine if a C++ NT makes it all the way here, we've got bigger problems

Reply:
Ah this is for when the test for the subclass tests get compiled and singleton symints start to show up in places. 
So technically a more accurate name for this PR is test_nestedtensor.